api/src/controllers/schedule.controller.ts

```diff
@@ -0,0 +1,89 @@
+import { ISchedule, Days } from '../interfaces/Schedule';
+import { create_schedule } from '../helpers/createMockData';
+import { IUser } from '../interfaces/User';
+
+const Schedule = require('../models/schedule.model');
+const User = require('../models/user.model');
+const jwt_decode = require('jwt-decode');
+
+/**
+ * Schedule Controller
+ *
+ * @export
+ * @class ScheduleController
+ */
+export class ScheduleController 
+{
+
+    /**
+     * Creates a schedule (currently mock)
+     * @return {*}
+     * @memberof ScheduleController
+     */
+    async create(): Promise<void> 
+    {
+        try
+        {
+            await create_schedule(10); //creates mock schedule
+        } catch (err) 
+        {
+            throw new Error('Error creating schedule.');
+        }
+    }
+
+    /**
+     * Retrieves the entire schedule that was previously created
+     * @param {string} authToken
+     * @return {*}  {Promise<ISchedule[]>}
+     * @memberof ScheduleController
+     */
+    async list(authToken: string): Promise<ISchedule[]>
+    {
+        let decoded_email = '';
```

Comment:
Or if you are only using the JWT to validate the USERS role is ADMIN. THen create a new function in Auth controller, that takes in the JWT token and returns if user is admin or not

Reply:
lol good point, it's way cleaner now. I'm doing authentication in routes now. 
Should be done.